### PR TITLE
fix: add jsdoc and do minor changes in utils.js

### PR DESCRIPTION
**Pull request by Uzlopak**
add jsdoc to the utils.js file to improve typings in other places.

also made minor changes, which i will comment

### CODE REVIEW

File: lib/core/util.js [hunk 1]
```diff
@@ -303,8 +369,12 @@ function destroy (stream, err) {
 }
 
 const KEEPALIVE_TIMEOUT_EXPR = /timeout=(\d+)/
+/**
+ * @param {string} val
+ * @returns {number | null}
+ */
 function parseKeepAliveTimeout (val) {
-  const m = val.toString().match(KEEPALIVE_TIMEOUT_EXPR)
+  const m = val.match(KEEPALIVE_TIMEOUT_EXPR)
```
Comment (Uzlopak) [inline lib/core/util.js:377]: parseKeepAliveTimeout gets according to my analysis always a string, so we can avoid the toString() call

File: lib/core/util.js [hunk 2]
```diff
@@ -363,17 +434,24 @@ function parseHeaders (headers, obj) {
...
 }
 
+/**
+ * @param {Buffer[]} headers
+ * @returns {string[]}
+ */
 function parseRawHeaders (headers) {
-  const len = headers.length
-  const ret = new Array(len)
+  const headersLength = headers.length
```
Comment (Uzlopak) [inline lib/core/util.js:442]: we can reuse the headersLength variable for the for loop

File: lib/core/util.js [hunk 3]
```diff
@@ -594,17 +700,24 @@ const headerCharRegex = /[^\t\x20-\x7e\x80-\xff]/
...
+ * Parse accordingly to RFC 9110
+ * @see https://www.rfc-editor.org/rfc/rfc9110#field.content-range
+ * @param {string} [range]
+ * @returns
+ */
 function parseRangeHeader (range) {
   if (range == null || range === '') return { start: 0, end: null, size: null }
 
-  const m = range ? range.match(/^bytes (\d+)-(\d+)\/(\d+)?$/) : null
+  const m = range ? range.match(rangeHeaderRegex) : null
```
Comment (Uzlopak) [inline lib/core/util.js:720]: extracted the range regex for better perf

File: lib/core/util.js [hunk 4]
```diff
@@ -614,20 +727,35 @@ function parseRangeHeader (range) {
...
   return obj
 }
 
+/**
+ * @param {Record<string|symbol, any>} obj
+ */
 function removeAllListeners (obj) {
-  for (const [name, listener] of obj[kListeners] ?? []) {
-    obj.removeListener(name, listener)
+  if (obj[kListeners] != null) {
```
Comment (Uzlopak) [inline lib/core/util.js:746]: if obj[kListeners] is not null or undefined, then we know it is an array, so only do the steps if we have somthing set.

**Review by mcollina — APPROVED** [review]
lgtm

**Review by metcoder95 — APPROVED** [review]

**Pull request merged**